src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -71,9 +71,17 @@ object LightGBMUtils {
   def getNumExecutors(dataset: Dataset[_], defaultListenPort: Int): Int = {
     val executors = getExecutors(dataset, defaultListenPort)
     if (executors.isEmpty) {
+      val master = dataset.sparkSession.sparkContext.master
+      if (master == "local") 1
       // Case when run locally in local[*]
-      val blockManager = BlockManagerUtils.getBlockManager(dataset)
-      blockManager.master.getMemoryStatus.size
+      else if (master.matches("local\\[[*\\d+]\\]")) {
+        val numCores = master.replace("local[", "").replace("]", "").split(",").head
```

Comment:
You can actually use regex match groups to just extract the element.

Reply:
done!